### ISSUE TITLE
Webpack v5.22.0` breaks HtmlWebpackLoader

### ISSUE THREAD

**Issue by trullock** [jantimon/html-webpack-plugin#1603]
See here, where webpack dont give a single sh*t: https://github.com/webpack/webpack/issues/12689

# Bug report

Updating from 5.21.2 to 5.22.0 makes my build spit out the following errors for all my `HtmlWebpackPlugins`:

`message:  Error: The loader "<path>\\node_modules\\html-webpack-plugin\\li  b\\loader.js!<path>\\src\\app\\app.html" didn't return html.\n+`

Relevant webpack config 

```
new HtmlWebpackPlugin({
			template: path.resolve(__dirname, "src", "app", "app.html"),
			templateParameters: {
				pages
			},
			filename: 'app.html',
			scriptLoading: "defer",
			chunks: ['app']
		})
```

rest of stack trace:

```
      '  - index.js:336 HtmlWebpackPlugin.evaluateCompilationResult\n' +
      '    [Useroo]/[html-webpack-plugin]/index.js:336:24\n' +        
      '  \n' +
      '  - index.js:243 \n' +
      '    [Useroo]/[html-webpack-plugin]/index.js:243:22\n' +        
      '  \n' +
      '  - runMicrotasks\n' +
      '  \n' +
      '  - task_queues.js:93 processTicksAndRejections\n' +
      '    internal/process/task_queues.js:93:5\n' +
      '  \n' +
      '  - async Promise.all\n' +
      '  \n' +
      '  - async Promise.all\n' +
      '  \n'
  },
```

**What is the expected behavior?**

Not to explode, it doesn't in 5.21.2 but does in 5.22.0

**Other relevant information:**
webpack version:  5.22.0 (not 5.21.2)
Node.js version:  12, 14
Operating System: Windows, Linux

**Comment by alexander-akait** [comment]
First and last warning for `sh*t`. Respect other developers.

**Comment by alexander-akait** [comment]
For the future, I recommend you - apply updates step by step, then it will be easier to determine where the problem came from

**Comment by trullock** [comment]
only updated webpack, from 5.21.2 to 5.22.0

**Comment by 3cp** [comment]
We confirm there is a breaking behaviour between webpack 5.21.2 and 5.22.0.
It manifested to a different error message for our setup.

```
Error: The loader "my-app/node_modules/html-webpack-plugin/lib/loader.js!/home/itghuc/my-app/index.ejs" didn't return html
```

Very basic usage for our setup.
```js
new HtmlWebpackPlugin({ template: 'index.ejs' })
```

**Comment by alexander-akait** [comment]
@trullock Please create reproducible test repo

**Comment by alexander-akait** [comment]
https://github.com/webpack/webpack/issues/12687#issuecomment-779530228, please update `html-webpack-plugin` to v5, v4 is not fully compatibility with webpack v5

**Comment by trullock** [comment]
> @trullock Please create reproducible test repo

Sure, I'll watch https://github.com/webpack/webpack/issues/12687#issuecomment-779530228 first and see what happens there.

I'll pursue updating html-webpack-plugin too

**Comment by jantimon** [comment]
@trullock does updating resolve the issue for you?

**Comment by ttttttssssss** [comment]
I have also encountered the above problems.
(What a bad day! I thought it was because of the environment.Because I deployed on Windows and developed on Mac)

There are two solutions:

1. update html-webpack-plugin@4.x.x -> @5.x.x
> https://github.com/webpack/webpack/issues/12689 
> `alexander-akait` :You need to update html-webpack-plugin to v5, v4 is not fully compatibility with webpack v5

or

2. webpack use < @5.22.0

**Comment by mixalistzikas** [comment]
We have the same problem. We have webpack 5.21.2 but html-webpack-plugin is dependency from something else. And we can't find the dependency. If we use 2 days ago package-lock.json it works. If we do clean npm install it breaks

**Comment by alexander-akait** [comment]
@mixalistzikas `npm ls html-webpack-plugin`

**Comment by mixalistzikas** [comment]
UNMET OPTIONAL DEPENDENCY @storybook/angular@6.1.18
  `-- UNMET OPTIONAL DEPENDENCY @storybook/core@6.1.18
    `-- UNMET OPTIONAL DEPENDENCY html-webpack-plugin@4.5.1

**Comment by alexander-akait** [comment]
@mixalistzikas run `npm i -D html-webpack-plugin`, you have `UNMET OPTIONAL DEPENDENCY`, it is bad

**Comment by mixalistzikas** [comment]
I moved storybook from optional to devDependencies and add `"html-webpack-plugin": "^5.1.0"`
Now I'm getting this on `npm ls html-webpack-plugin`

    ``-- (empty)`

Some problem. 

Error: Cannot read property 'tapAsync' of undefined

**Comment by alexander-akait** [comment]
You forget to run `npm i`

**Comment by mixalistzikas** [comment]
I did clean install again. Delete node_modules, delete package-lock.json and npm install

**Comment by alexander-akait** [comment]
`npm ls` should show `html-webpack-plugin`, if you don't see it you do something wrong

**Comment by mixalistzikas** [comment]
My bad... I'm removing my last comments... I will try again your first suggestion

**Comment by mixalistzikas** [comment]
+-- @storybook/angular@6.1.18
| `-- @storybook/core@6.1.18
|   `-- html-webpack-plugin@4.5.1
`-- html-webpack-plugin@5.1.0

**Comment by alexander-akait** [comment]
@mixalistzikas All works fine?

**Comment by mixalistzikas** [comment]
No, I still get errors
![image](https://user-images.githubusercontent.com/562128/108212389-7b8ddf00-7136-11eb-8271-8da6f9879ac7.png)

**Comment by alexander-akait** [comment]
As you can see the problem in `script-webpack-plugin` from `@angular-devkit`, nothing to fix here

**Comment by MartB** [comment]
@mixalistzikas just add the following into your project package.json and hope that its compatible^^

```
  "resolutions": {
    "html-webpack-plugin": "^5.1.0"
  }
```

**Comment by 913453448** [comment]
you can do this
<img width="1605" alt="屏幕快照 2021-02-18 上午11 09 12" src="https://user-images.githubusercontent.com/15084079/108299704-dd535500-71d9-11eb-8627-b94af730485c.png">

**Comment by zhangbowen-github** [comment]
Upgrading html-webpack-plugin from 4.5.1 to 5.1.0 fixed the issue for us.  😀👍

**Comment by takhello** [comment]
Really a dangerous update, sorry

**Comment by jantimon** [comment]
I released a fix as `html-webpack-plugin@4.5.2` - however I strongly recommend to use webpack 5 with html-webpack-plugin 5

**Comment by trullock** [comment]
That's fantastic, thanks. I wasn't using v5 originally due to some other bug which is perhaps now fixed separately.  4.5.2 hopefully gets us out of this scrape and I'll come back if there's an issue with the current V5

**Comment by dasler-fw** [comment]
> I released a fix as `html-webpack-plugin@4.5.2` - however I strongly recommend to use webpack 5 wit `html-webpack-plugin@5`

It's only thing that finally helped: using @4.5.2.
HtmlWebpackPlugin is not working with webpack@5.21.0 and newer.
